pkgs/development/libraries/qrupdate/default.nix

```diff
@@ -3,25 +3,27 @@
 , gfortran
 , blas
 , lapack
+, which
 }:
-stdenv.mkDerivation {
-  name = "qrupdate-1.1.2";
+
+stdenv.mkDerivation rec {
+  pname = "qrupdate";
+  version = "1.1.2";
   src = fetchurl {
-    url = "mirror://sourceforge/qrupdate/qrupdate-1.1.2.tar.gz";
+    url = "mirror://sourceforge/qrupdate/${pname}-${version}.tar.gz";
     sha256 = "024f601685phcm1pg8lhif3lpy5j9j0k6n0r46743g4fvh8wg8g2";
   };
 
-  configurePhase =
-    ''
-      export PREFIX=$out
-      sed -i -e 's,^BLAS=.*,BLAS=-L${blas}/lib -lblas,' \
-          -e 's,^LAPACK=.*,LAPACK=-L${lapack}/lib -llapack,' \
-          Makeconf
-    ''
-    + stdenv.lib.optionalString blas.isILP64
-    ''
-      sed -i Makeconf -e '/^FFLAGS=.*/ s/$/-fdefault-integer-8/'
-    '';
+  makeFlags = assert (blas.isILP64 == lapack.isILP64); [
+    "LAPACK=-L${lapack}/lib"
+    "BLAS=-L${blas}/lib"
+    "PREFIX=$out"
+  ]
+    ++ stdenv.lib.optionals blas.isILP64 [
+      "FFLAGS=-fdefault-integer-8"
+    ]
+  ;
+  NIX_LDFLAGS = "-lblas -llapack";
```

Comment:
This is unfortunate, that `makeFlags` are space-segmented. We really need structureAttrs.

For now, could you add the `makeFlags` containing spaces to `makeFlagsArray` in `preBuild`?

Reply:
> For now, could you add the makeFlags containing spaces to makeFlagsArray in preBuild?

That worked. Using `makeFlagsArray` as a Nix array didn't. So we prefer setting `makeFlagsArray` in `preBuild` instead of using `makeFlags` and `NIX_LDFLAGS`? The later feels less bashy and more Nixy.

Reply:
Yeah, I prefer more correct bashiness that can be replaced by nice Nixiness with structuredAttrs to less correct Nixiness that will likely be forgotten about.